ts/src/bingx.ts

```diff
@@ -0,0 +1,2342 @@
+                'rate': undefined,
+            },
+        };
+    }
+
+    parseTransactionStatus (status) {
+        const statuses = {
+            '0': 'ok',
+            '10': 'pending',
+            '20': 'rejected',
+            '30': 'ok',
+            '40': 'rejected',
+            '50': 'ok',
+            '60': 'pending',
+            '70': 'rejected',
+            '2': 'pending',
+            '3': 'rejected',
+            '4': 'pending',
+            '5': 'rejected',
+            '6': 'ok',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    async setMarginMode (marginMode: string, symbol: string = undefined, params = {}) {
+        /**
+         * @method
+         * @name bingx#setMarginMode
+         * @description set margin mode to 'cross' or 'isolated'
+         * @see https://bingx-api.github.io/docs/#/swapV2/trade-api.html#Switch%20Margin%20Mode
+         * @param {string} marginMode 'cross' or 'isolated'
+         * @param {string} symbol unified market symbol
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @returns {object} response from the exchange
+         */
+        this.checkRequiredSymbol ('setMarginMode', symbol);
+        marginMode = marginMode.toUpperCase ();
+        if (marginMode === 'CROSS') {
+            marginMode = 'CROSSED';
+        }
+        if (marginMode !== 'ISOLATED' && marginMode !== 'CROSSED') {
+            throw new BadRequest (this.id + ' setMarginMode() marginMode argument should be isolated or cross');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        if (market['type'] !== 'swap') {
+            throw new BadSymbol (this.id + ' setMarginMode() supports swap contracts only');
+        }
+        const request = {
+            'symbol': market['id'],
+            'marginType': marginMode,
+        };
+        return await this.swapV2PrivatePostTradeMarginType (this.extend (request, params));
+    }
+
+    async setMargin (symbol: string, amount, params = {}) {
+        /**
+         * @method
+         * @name bingx#setMargin
+         * @description Either adds or reduces margin in an isolated position in order to set the margin to a specific value
+         * @see https://bingx-api.github.io/docs/swapV2/trade-api.html#_15-adjust-isolated-margin
+         * @param {string} symbol unified market symbol of the market to set margin in
+         * @param {float} amount the amount to set the margin to
+         * @param {object} params parameters specific to the bingx api endpoint
+         * @returns {object} A [margin structure]{@link https://docs.ccxt.com/#/?id=add-margin-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const type = this.safeInteger (params, 'type'); //  1 increase margin 2 decrease margin
+        this.checkRequiredArgument ('setMargin', type, 'type');
```

Comment:
here we might throw a customized exception explaining what type is and the values accepted